filters/openpolicyagent/internal/envoy/skipperadapter.go

```diff
@@ -12,6 +12,10 @@ import (
 
 func AdaptToExtAuthRequest(req *http.Request, metadata *ext_authz_v3_core.Metadata, contextExtensions map[string]string, rawBody []byte) (*ext_authz_v3.CheckRequest, error) {
 
+	if !utf8.ValidString(req.URL.Path) {
```

Comment:
This maintains the current behaviour, but shouldn't this check the `RequestURI()` as well?

Reply:
`RequestURI()` returns the encoded path, which bypasses our validation.
`req.URL.Path` returns the decoded path. Therefore I decided to use it to preserve the current behaviour.

Reply:
Ok, that makes sense but we will miss invalid UTF8 in the query the way it is written now.

Reply:
RequestURI() is likely enforced by Go stdlib net/http to be compliant with RFC, so we will not have the issue there.